servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/PipelinedStreamingHttpConnection.java

```diff
@@ -40,16 +41,40 @@
     @Override
     protected Publisher<Object> writeAndRead(Publisher<Object> requestStream,
                                              @Nullable final FlushStrategy flushStrategy) {
-        if (flushStrategy == null) {
-            return connection.write(requestStream);
-        } else {
-            // TODO(scott): if we can remove the flush state on the connection we can simplify the control flow here.
-            return Publisher.defer(() -> {
+        assert connectionContext().protocol().major() <= 1 : "Unexpected protocol version";
+        return (flushStrategy == null ? connection.write(requestStream) :
+            Publisher.defer(() -> {
                 final Cancellable resetFlushStrategy = connection.updateFlushStrategy(
                         (prev, isOriginal) -> isOriginal ? flushStrategy : prev);
                 return connection.write(requestStream, connection::defaultFlushStrategy,
                         WriteDemandEstimators::newDefaultEstimator).afterFinally(resetFlushStrategy::cancel);
+            })).beforeFinally(new TerminalSignalConsumer() {
+                @Override
+                public void onComplete() {
+                    // noop
+                }
+
+                @Override
+                public void onError(final Throwable throwable) {
+                    // noop
+                }
+
+                @Override
+                public void cancel() {
+                    // If the HTTP/1.x request gets cancelled, we pessimistically assume that the transport will close
+                    // the connection since the Subscriber did not read the entire response and cancelled. This reduces
+                    // the time window during which a connection is eligible for selection by the load balancer post
+                    // cancel and the connection being closed by the transport.
+                    // Transport MAY not close the connection if cancel raced with completion and completion was seen by
+                    // the transport before cancel. We have no way of knowing at this layer if this indeed happen.
+                    closeAsync().subscribe();
```

Comment:
identifying the rational for closure is one of the most common support questions. can we add an internal utility method shared by other cases that force closure to provide additional visibility?

```
static void doClose(AsyncClosable closable, String reason) {
  closable.closeAsync();
  LOGGER.debug("closing {} because {}", closable, reason):
}